### PR TITLE
Added an edit button and an end of page button to the main menu

### PR DESCRIPTION
**Pull request by NinaHerrmann**
The edit button is visible for each page which allows editing (assuming the current user has the correct priviledges).
The end of page button is shown on all pages beside the login page.

### CODE REVIEW

**Comment by NinaHerrmann** [comment]
@tobiasreischmann @Dagefoerde If you have time until tomorrow evening a review would be nice.

File: classes/output/core_renderer.php [hunk 1]
```diff
@@ -131,6 +131,170 @@ public function main_menu() {
...
+     */
+    public function navbar_plugin_output() {
+        $output = '';
+        if ($this->page->user_allowed_editing()) {
+            $output .= $this->add_edit_button();
+        }
+        if (!$this->is_login_page()) {
+            $output .= $this->add_endofpage_button();
+        }
+        // Give subsystems an opportunity to inject extra html content. The callback
```
Comment (tobiasreischmann) [inline classes/output/core_renderer.php]: Nutze hier lieber parent::navbar_plugin_output() anstatt den Code des Parent zu kopieren.
Anschließend kannst du einen output davorhängen.

**Comment by Dagefoerde** [comment]
> For what ever reasons, there is a little triangle in the menu, which comes from the go to bottom button:
> image

Because it has the class `popover-region-toggle`  (which is not needed, because there is neither a popover, nor a toggle).

File: classes/output/core_renderer.php [hunk 2]
```diff
@@ -131,6 +131,170 @@ public function main_menu() {
...
+        $html .= \html_writer::div($tag, 'popover-region-toggle nav-link');
+        return $html;
+    }
+
+    /**
+     * Checks the current pagetype and sets the edit param based on the pagetype.
+     * TODO: n_herr03 the switch statement is not exhaustive. Moreover, not all cases are tested.
+     * @return object url with the required edit options
+     */
+    private function switch_pagetype() {
```
Comment (Dagefoerde) [inline classes/output/core_renderer.php]: Can we use a more speaking name, such as `get_edit_button_url_by_pagetype`?

Also, the switch is exhaustive: There is a `default` clause at the end.

File: classes/output/core_renderer.php (hunk 1 above)
Comment (NinaHerrmann) [reply classes/output/core_renderer.php]: https://github.com/learnweb/moodle-theme_wwu2019/commit/80d72732cf113d9574a926386386eabd942b6bec

File: classes/output/core_renderer.php (hunk 2 above)
Comment (NinaHerrmann) [reply classes/output/core_renderer.php]: https://github.com/learnweb/moodle-theme_wwu2019/commit/8bb0ad82a8a13a84b8da70d60f5102055a5d2ee2 and https://github.com/learnweb/moodle-theme_wwu2019/commit/f03d30f29764baebdbc7925fb99fea3bea57a721

**Comment by NinaHerrmann** [comment]
I removed the `popover-region-toggle` class in https://github.com/learnweb/moodle-theme_wwu2019/commit/d92073dd34ba2a976936b3137d1d1a10f5e63eda.

File: classes/output/core_renderer.php (hunk 1 above)
Comment (NinaHerrmann) [reply classes/output/core_renderer.php]: @tobiasreischmann approved?

**Pull request merged**